darts/preprocessing/base_transformer.py

```diff
@@ -0,0 +1,32 @@
+
+from typing import TypeVar, Generic, Callable
+from darts.logging import raise_if_not, get_logger
+
+logger = get_logger(__name__)
+T = TypeVar('T')
+
+
+class BaseTransformer(Generic[T]):
+    def __init__(self):
+        self.validators = [
+            lambda x: True
+        ]
+
+    def add_validator(self, validator: Callable[[T], bool]):
```

Comment:
Would it be possible to consider specifying the validators when building the class (as arguments of `__init__()`? This way it would be easier to see Transformers as immutable things that are not modified once built :)

Reply:
Of course, its not hard to do and if we agree on this (making transformer immutable) then add_validator would return new transformer. +1 for specifying validators in init.